csrc/fusion_segmenter.cpp

```diff
@@ -3399,13 +3399,21 @@ void SegmentCandidateFinder::forwardInputs() {
         continue;
       }
 
-      if (expr->output(0)->uses().size() > 1) {
+      // expr is a unary op so there is a single output. Here we look at that
+      // output's further uses
+      auto output_uses = expr->output(0)->uses();
+
+      if (output_uses.size() == 1) {
+        // If there is a single use, visit it to try and extend the chain of
+        // unaryOps
+        to_visit.emplace_back(output_uses[0]);
```

Comment:
Not going to happen anymore, but just in case:

```suggestion
        to_visit.emplace_back(output_uses.at(0));
```